src/main/java/org/springframework/integration/kafka/listener/KafkaNativeOffsetManager.java

```diff
@@ -0,0 +1,235 @@
+import org.springframework.integration.kafka.core.ConsumerException;
+import org.springframework.integration.kafka.core.DefaultConnectionFactory;
+import org.springframework.integration.kafka.core.KafkaConsumerDefaults;
+import org.springframework.integration.kafka.core.Partition;
+import org.springframework.integration.kafka.core.PartitionNotFoundException;
+import org.springframework.integration.kafka.core.Result;
+import org.springframework.integration.kafka.support.ZookeeperConnect;
+import org.springframework.retry.RetryCallback;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.listener.RetryListenerSupport;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+
+import com.gs.collections.impl.factory.Maps;
+
+import kafka.client.ClientUtils$;
+import kafka.common.ErrorMapping;
+import kafka.common.OffsetAndMetadata;
+import kafka.network.BlockingChannel;
+import kafka.utils.ZKStringSerializer$;
+
+/**
+ * Implementation of an {@link OffsetManager} that uses kafka native 'topic' offset storage.
+ *
+ * @author Chris Lemper
+ * @author Marius Bogoevici
+ */
+public class KafkaNativeOffsetManager extends AbstractOffsetManager implements InitializingBean {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(KafkaNativeOffsetManager.class);
+
+	private static final String PARTITION_ATTRIBUTE = "partition";
+
+	private final Map<Partition, BrokerAddress> offsetManagerBrokerAddressCache = new ConcurrentHashMap<>();
+
+	private RetryTemplate retryTemplate;
+
+	private ZkClient zkClient;
+
+	/**
+	 * @param connectionFactory a Kafka connection factory
+	 * @param zookeeperConnect zookeeper connection for retrieving 
+	 * @param initialOffsets a map of partitions to initial offsets
+	 */
+	public KafkaNativeOffsetManager(ConnectionFactory connectionFactory, ZookeeperConnect zookeeperConnect,
+									Map<Partition, Long> initialOffsets) {
+		super(connectionFactory, initialOffsets);
+		final String zkServers = zookeeperConnect.getZkConnect();
+		zkClient = new ZkClient(zkServers, 
+				Integer.parseInt(zookeeperConnect.getZkSessionTimeout()), 
+				Integer.parseInt(zookeeperConnect.getZkConnectionTimeout()),
+				ZKStringSerializer$.MODULE$);
+	}
+
+	/**
+	 * @param connectionFactory a Kafka connection factory
+	 * @param zookeeperConnect the zookeeper connection information
+	 * @param consumerId the kafka consumer ID
+	 * @param referenceTimestamp the reset timestamp for initial offsets
+	 */
+	public KafkaNativeOffsetManager(ConnectionFactory connectionFactory, ZookeeperConnect zookeeperConnect,
+									String consumerId, long referenceTimestamp) {
+		this(connectionFactory, zookeeperConnect, Collections.<Partition, Long>emptyMap());
+		setConsumerId(consumerId);
+		setReferenceTimestamp(referenceTimestamp);
```

Comment:
Minor, of course, but I wonder if we really should have these two options as ctor args.
Maybe that one for the `connectionFactory` and `zookeeperConnect` would be enough ?